src/wkwebview/drag_drop.rs

```diff
@@ -2,179 +2,147 @@
-unsafe fn get_handler(this: &Object) -> &mut Box<dyn Fn(DragDropEvent) -> bool> {
-  let delegate: *mut c_void = *this.get_ivar(DRAG_DROP_HANDLER_IVAR);
-  &mut *(delegate as *mut Box<dyn Fn(DragDropEvent) -> bool>)
-}
+static OBJC_PERFORM_DRAG_OPERATION: Lazy<
+  extern "C" fn(&NSView, Sel, &ProtocolObject<dyn NSDraggingInfo>) -> objc2::runtime::Bool,
+> = Lazy::new(|| unsafe {
+  std::mem::transmute(
+    class!(WKWebView)
+      .instance_method(objc2::sel!(performDragOperation:))
+      .unwrap()
+      .implementation(),
+  )
+});
 
-unsafe fn collect_paths(drag_info: id) -> Vec<PathBuf> {
-  use cocoa::{
-    appkit::{NSFilenamesPboardType, NSPasteboard},
-    foundation::{NSFastEnumeration, NSString},
-  };
+static OBJC_DRAGGING_UPDATED: Lazy<
+  extern "C" fn(&NSView, Sel, &ProtocolObject<dyn NSDraggingInfo>) -> NSDragOperation,
+> = Lazy::new(|| unsafe {
+  std::mem::transmute(
+    class!(WKWebView)
+      .instance_method(objc2::sel!(draggingUpdated:))
+      .unwrap()
+      .implementation(),
+  )
+});
 
-  let pb: id = msg_send![drag_info, draggingPasteboard];
+pub(crate) unsafe fn collect_paths(drag_info: &ProtocolObject<dyn NSDraggingInfo>) -> Vec<PathBuf> {
+  let pb = drag_info.draggingPasteboard();
   let mut drag_drop_paths = Vec::new();
-  let types: id = msg_send![class!(NSArray), arrayWithObject: NSFilenamesPboardType];
-  if !NSPasteboard::availableTypeFromArray(pb, types).is_null() {
-    for path in NSPasteboard::propertyListForType(pb, NSFilenamesPboardType).iter() {
-      drag_drop_paths.push(PathBuf::from(
-        CStr::from_ptr(NSString::UTF8String(path))
-          .to_string_lossy()
-          .into_owned(),
-      ));
+  let types = NSArray::arrayWithObject(NSFilenamesPboardType);
+
+  if pb.availableTypeFromArray(&types).is_some() {
+    let paths = pb.propertyListForType(NSFilenamesPboardType).unwrap();
+    let paths: Id<NSArray<NSString>> = Id::<AnyObject>::cast(paths.clone());
+    for path in paths.to_vec() {
+      let path = CStr::from_ptr(path.UTF8String()).to_string_lossy();
+      drag_drop_paths.push(PathBuf::from(path.into_owned()));
     }
   }
   drag_drop_paths
 }
 
-extern "C" fn dragging_updated(this: &mut Object, sel: Sel, drag_info: id) -> NSDragOperation {
-  let dl: NSPoint = unsafe { msg_send![drag_info, draggingLocation] };
-  let frame: NSRect = unsafe { msg_send![this, frame] };
+pub(crate) fn dragging_entered(
+  this: &WryWebView,
+  drag_info: &ProtocolObject<dyn NSDraggingInfo>,
+) -> NSDragOperation {
+  let paths = unsafe { collect_paths(drag_info) };
+  let dl: NSPoint = unsafe { drag_info.draggingLocation() };
+  let frame: NSRect = this.frame();
+  let position = (dl.x as i32, (frame.size.height - dl.y) as i32);
+
+  let listener = &this.ivars().drag_drop_handler;
+  if !listener(DragDropEvent::Enter { paths, position }) {
+    // Reject the Wry file drop (invoke the OS default behaviour)
+    OBJC_DRAGGING_ENTERED(this, objc2::sel!(draggingEntered:), drag_info)
```

Comment:
I don't think you need to change it in this PR, but I do think you could consider using Objective-C's built-in mechanism for this kind of stuff at some point, namely sending a message to the super class:
```suggestion
    unsafe { msg_send![super(this), draggingEntered: drag_info] }
```

Same goes below.